docs/frontend/40-git.rst

```diff
@@ -1,22 +1,133 @@
+Also, the bigger a PR is, the longer it takes to re-review after requested changes are carried out.
 
-.. todo:: Pull requests
+When doing indentation changes, *never* combine this with code changes since it will be impossible to review.
+
+When the code *touches* a code block or line that doesn't conform to coding style guidelines, use the opportunity
+to adjust that/those lines. (ES6 arrow functions instead of ``function () {}.bind(this)`` for example.)
+It could also include adjacent lines that will be included in the diff output.
+This does *not* mean to find all occurrences in the file and replacing those.
+
+Reviewing a PR
+--------------
+
+When reviewing a PR, as a guideline, consider the following for the new code:
+
+* Is the code readable? If not, the preferred approach is to refactor it to be self-documenting, otherwise add comments.
+* Is it performant?
+* Does it conform to coding style guidelines?
+* What kind of WCT tests could/should be designed around this?
+* Are there any related scenarios that could be affected by this? This should be tested out.
+
+.. todo:: More guidelines
+
+.. note::
+  On the "Conversation" tab of the PR, next to the merge-button, there are instructions on
+  how to check out and test the PR locally with the CLI.
 
 .. seealso::
-    :ref:`github-git`
\ No newline at end of file
+  https://help.github.com/articles/about-pull-request-reviews/
+  https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/
+
+Merging a PR
+------------
+
+If you are the only, or last, reviewer of a PR, you should also merge the PR.
+
+:guilabel:`Rebase and merge` will add the commits directly on master.
+This is a good starting point and should be used for any PR only containing one commit.
+If there are more than one commit, they should make sense to have in the commit history individually.
+
+:guilabel:`Squash and merge` will squash all commits into one.
+This is a good approach for a PR that contains detail changes or a minor feature but has
+grown in number of commits, maybe due to number of requested changes during review.
+If it feels like the commits mostly reference the same thing and the individual commits would
+add more noise than traceability to the git history, this would be a good candidate.
+
+:guilabel:`Create a merge commit` will create a merge commit along with all the commits in the PR.
+This is the default git behavior and makes sense if the PR is a bigger feature with several parts,
+so they should be kept together as they affect the same feature, but they also make sense individually.
+
+Deployment and branches
+~~~~~~~~~~~~~~~~~~~~~~~
+
+
+Artifacts / binaries
+--------------------
+
+Once the build promotion pipeline (integration -> alpha -> beta -> prod) has started, we
+need a way to guarantee consistent builds across environments.
+
+.. todo:: Define a way to do this - copy integration build output files to different environments?
```

Comment:
I've discussed this a bit with @jokedst some time ago, and a pretty good solution would be to build the app for all the environments, and store these builds as zip with the release in github.

Then deploying would mean fetching the right zip file and sending it to the right environment.

With this method, we are sure to run the app with the exact same dependencies.

Also, reverting a deploy would be simpler as well.

Reply:
Yes that sounds good but the zips from GitHub are just the repo at a specific time? Or how can we manipulate the zips to include our build/dependencies?
But we can make our own service for this, I guess?